Neo4jClient.Full.Shared/Transactions/Bolt/BoltTransactionPromotableSinglePhasesNotification.cs

```diff
@@ -30,13 +30,19 @@ public byte[] Promote()
             var driverTx = transaction.DriverTransaction;
             var session = transaction.Session;
             transaction = null;
-            
-            return ResourceManager.Promote(new BoltTransactionExecutionEnvironment(client.ExecutionConfiguration)
+
+            // BUG: .NET 4.7.1 introduced a bug where the current transaction wouldn't get restored
```

Comment:
With this - when it's fixed in the framework - will this need to change as well - or are we good regardless?

Reply:
It's not sure it will ever be "fixed". A "fix" was created for 4.7.2, but that broke the case where TransactionScope + TransactionFlowAsyncOption.Enabled + Remote call was used *without* crossing AppDomain boundaries. So that "fix" was changed for 4.8 so the fix has been made opt-in as it breaks stuff.

In addition to this workaround, MS proposed two other fixes, but they both seemed to have side-effects and more difficult to understand.

So it might never be fixed, and at least not fixed until much later, and the fix might actually be just shuffling the bugs around :/

Reply:
You can see the response with workarounds here: https://github.com/Microsoft/dotnet-framework-early-access/issues/7#issuecomment-435439034

Reply:
I'll add this discussion in the comment as well

Reply:
NM, I see I already link the issue in the comment.

Reply:
Btw.. What we can do is change the restoring to only restore if the transaction is lost. That way iff .NET fixes this issue, we won't set the transaction, and thus hopefully avoid any possible side-effects.

Reply:
OK, that sounds like a good solution - that way if it is fixed we're less worried about hidden bugs

Reply:
Done